go/ethclient/mgmtcontractlib/mgmt_contract_lib.go

```diff
@@ -183,6 +190,22 @@ func (c *contractLibImpl) CreateInitializeSecret(tx *obscurocommon.L1InitializeS
 	}
 }
 
+func (c *contractLibImpl) GetHostAddresses() (ethereum.CallMsg, error) {
+	data, err := c.contractABI.Pack(GetHostAddressesMethod)
+	if err != nil {
+		return ethereum.CallMsg{}, fmt.Errorf("could not pack the call data. Cause: %w", err)
+	}
+	return ethereum.CallMsg{To: c.addr, Data: data}, nil
+}
+
+func (c *contractLibImpl) DecodeCallResponse(callResponse []byte) ([]interface{}, error) {
```

Comment:
Can this return a []string ?

Reply:
Not sure yet, haven't used it in a wide enough set of scenarios. I'm tempted to leave as interface for now, and we can narrow it down later. Unless you're sure it's always []string?

Reply:
Fixed